### PR TITLE
[IOTDB-938] Re-implement Gorilla encoding algorithm

### PR DESCRIPTION
**Pull request by SteveYurongSu**
JIRA: https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-938

### CODE REVIEW

**Comment by SteveYurongSu** [comment]
Performance report: https://cwiki.apache.org/confluence/display/IOTDB/IOTDB-938+Re-implement+Gorilla+encoding+algorithm

**Comment by HTHou** [comment]
https://github.com/apache/iotdb/pull/1856/checks?check_run_id=1306787563
Nice work! I invoked the sonar cloud for this pr, please check it. :)

**Comment by SteveYurongSu** [comment]
> https://github.com/apache/iotdb/pull/1856/checks?check_run_id=1306787563
> Nice work! I invoked the sonar cloud for this pr, please check it. :)

Done! Most code smells are from the legacy code, I've fixed them. Please have a check again :)

File: tsfile/src/main/java/org/apache/iotdb/tsfile/encoding/decoder/IntGorillaDecoder.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    readNext(in);
+    if (storedValue == GORILLA_ENCODING_ENDING_INTEGER) {
+      hasNext = false;
+    }
+    return storedValue;
+  }
+
+  @SuppressWarnings("squid:S128")
+  protected int readNext(ByteBuffer in) {
+    int controlBits = readNextClearBit(2, in);
```
Comment (HTHou) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/encoding/decoder/IntGorillaDecoder.java:69]: This `controlBits` can only be 0, 2 and 3, right? How do you think change it and the return type of method `readNextClearBit` to byte?

Comment (SteveYurongSu) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/encoding/decoder/IntGorillaDecoder.java]: You're right. I've changed it.

**Review by HTHou — APPROVED** [review]
LGTM

**Review by samperson1997 — APPROVED** [review]
Hi @SteveYurongSu , excellent work and impressive experiment result. Just one question: when will we remove `GORILLA_V1`? And should we change `GORILLA_V2` to `GORILLA` at that time?

**Comment by SteveYurongSu** [comment]
> Hi @SteveYurongSu , excellent work and impressive experiment result. Just one question: when will we remove `GORILLA_V1`? And should we change `GORILLA_V2` to `GORILLA` at that time?

Thanks!

`GORILLA_V1` can be completely replaced by `GORILLA_V2`. I think we can delete the code of `GORILLA_V1` and change `GORILLA_V2` to `GORILLA` in a new release version, but the premise is that we need to make a version upgrade tool.

**Review by samperson1997 — APPROVED** [review]

**Pull request merged**